internal/pkg/agent/application/monitoring/v1_monitor.go

```diff
@@ -325,133 +311,86 @@ func (b *BeatsMonitor) injectLogsInput(cfg map[string]interface{}, componentIDTo
-					"inactive": "5m",
-				},
-			},
-			"parsers": []interface{}{
-				map[string]interface{}{
-					"ndjson": map[string]interface{}{
-						"message_key":    "message",
-						"overwrite_keys": true,
-						"add_error_key":  true,
-						"target":         "",
 					},
 				},
-			},
-			"processors": []interface{}{
+				// copy component.dataset as the real dataset
 				map[string]interface{}{
-					"add_fields": map[string]interface{}{
-						"target": "data_stream",
-						"fields": map[string]interface{}{
-							"type":      "logs",
-							"dataset":   fmt.Sprintf("elastic_agent.%s", fixedBinaryName),
-							"namespace": monitoringNamespace,
+					"copy_fields": map[string]interface{}{
+						"fields": []interface{}{
+							map[string]interface{}{
+								"from": "component.dataset",
+								"to":   "data_stream.dataset",
+							},
 						},
+						"fail_on_error":  false,
+						"ignore_missing": true,
 					},
 				},
+				// possible it's a log message from agent itself (doesn't have component.dataset)
 				map[string]interface{}{
-					"add_fields": map[string]interface{}{
-						"target": "event",
-						"fields": map[string]interface{}{
-							"dataset": fmt.Sprintf("elastic_agent.%s", fixedBinaryName),
+					"copy_fields": map[string]interface{}{
+						"fields": []interface{}{
+							map[string]interface{}{
+								"from": "data_stream.dataset_original",
+								"to":   "data_stream.dataset",
+							},
 						},
+						"fail_on_error": false,
 					},
 				},
+				// drop the original dataset copied and the event.dataset (as it will be updated)
 				map[string]interface{}{
-					"add_fields": map[string]interface{}{
-						"target": "elastic_agent",
-						"fields": map[string]interface{}{
-							"id":       b.agentInfo.AgentID(),
-							"version":  b.agentInfo.Version(),
-							"snapshot": b.agentInfo.Snapshot(),
+					"drop_fields": map[string]interface{}{
+						"fields": []interface{}{
+							"data_stream.dataset_original",
+							"event.dataset",
 						},
 					},
 				},
+				// update event.dataset with the now used data_stream.dataset
 				map[string]interface{}{
-					"add_fields": map[string]interface{}{
-						"target": "agent",
-						"fields": map[string]interface{}{
-							"id": b.agentInfo.AgentID(),
+					"copy_fields": map[string]interface{}{
+						"fields": []interface{}{
+							map[string]interface{}{
+								"from": "data_stream.dataset",
+								"to":   "event.dataset",
+							},
 						},
 					},
 				},
+				// coming from logger, added by agent (drop)
 				map[string]interface{}{
 					"drop_fields": map[string]interface{}{
 						"fields": []interface{}{
-							"ecs.version", //coming from logger, already added by libbeat
+							"ecs.version",
 						},
 						"ignore_missing": true,
 					},
 				},
-			},
-		})
+				// adjust destination data_stream based on the data_stream fields
+				map[string]interface{}{
+					"add_formatted_index": map[string]interface{}{
```

Comment:
Are the destination datastreams all the same as they were before this change? Are there any new datastreams created here?

I haven't tested this locally yet, I just want to do an early check that we haven't introduced anything that could be considered a breaking change.

Reply:
The resulting datastreams are different in that now they are grouped by spawned component. This is done on purpose and a design decision of V2. The allows logs from each component that now has its own status to be grouped properly. So you can correlate status of a component to the exact logs of that component.

Reply:
👍